### PR TITLE
WIP: Make TarantoolCursor implement Iterator

### PR DESCRIPTION
**Pull request by akudiyar**
These changes are necessary for correct Spark connector implementation and so on

### CODE REVIEW

File: src/main/java/io/tarantool/driver/api/cursor/TarantoolCursorBase.java [hunk 1]
```diff
@@ -48,7 +48,7 @@ public boolean next() throws TarantoolClientException {
     }
 
     @Override
-    public T get() throws TarantoolSpaceOperationException {
+    public T next() throws TarantoolSpaceOperationException {
         if (getCurrentValue() == null) {
             throw new TarantoolSpaceOperationException(
```
Comment (vrogach2020) [inline src/main/java/io/tarantool/driver/api/cursor/TarantoolCursorBase.java:53]: Iterators throw NoSuchElementException when no data avalilable

File: src/main/java/io/tarantool/driver/api/cursor/TarantoolCursorBase.java [hunk 2]
```diff
@@ -39,7 +39,7 @@ protected static long calcLimit(long conditionLimit, long batchSize, long spaceO
     }
 
     @Override
-    public boolean next() throws TarantoolClientException {
+    public boolean hasNext() throws TarantoolClientException {
```
Comment (vrogach2020) [inline src/main/java/io/tarantool/driver/api/cursor/TarantoolCursorBase.java:42]: Logic of this method must be changed - hasNext() must not change the state of iterator (see advanceIterator()).

File: src/main/java/io/tarantool/driver/api/cursor/TarantoolCursor.java [hunk 3]
```diff
@@ -14,19 +15,5 @@
...
-     * @throws TarantoolClientException if the request to server failed.
-     */
-    boolean next() throws TarantoolClientException;
-
-    /**
-     * @return current element or null if cursor is not initialized
-     * @throws TarantoolSpaceOperationException when no data is available
-     */
-    T get() throws TarantoolSpaceOperationException;
+public interface TarantoolCursor<T extends Packable> extends Iterator<T> {
```
Comment (vrogach2020) [inline src/main/java/io/tarantool/driver/api/cursor/TarantoolCursor.java:18]: Cursors were implemented looking at jdbc cursor. Iterators have different behavior and this implemetation will need significant changes. Maybe it's better to wrap this class [ResultSetIterator](http://commons.apache.org/proper/commons-dbutils/apidocs/org/apache/commons/dbutils/ResultSetIterator.html)  for example.

**Review by vrogach2020 — CHANGES_REQUESTED** [review]

**Comment by akudiyar** [comment]
Closing the PR considering the above comments. Implementing java's Iterator is not necessary for the Spark connector.

**Pull request closed without merging**